components/sup/src/manager/service/supervisor.rs

```diff
@@ -235,17 +242,19 @@ impl Supervisor {
 
     /// Create a PID file for a running service
     fn create_pidfile(&self) -> Result<()> {
-        match self.pid {
-            Some(pid) => {
-                debug!("Creating PID file for child {} -> {}",
-                       self.pid_file.display(),
-                       pid);
-                let mut f = File::create(&self.pid_file)?;
-                write!(f, "{}", pid)?;
-                Ok(())
-            }
-            None => Ok(()),
+        if let Some(pid) = self.pid {
+            debug!("Creating PID file for child {} -> {}",
+                   self.pid_file.display(),
+                   pid);
+            let w = AtomicWriter::new(&self.pid_file)?;
+            w.with_writer(|f| {
+                 write!(f, "{}", pid)?;
+                 Ok(())
+             })
```

Comment:
I think you can save a couple lines by replacing L249–253 with
```rust
            fs::atomic_write(&self.pid_file, pid.to_string())
```

Reply:
Oh, nice.  That's much better.